File: AD-HOC/PYTHON/1026_1147/1147.py
dr = [-1, -2, -2, -1, 1, 2, 2, 1]
dc = [-2, -1, 1, 2, 2, 1, -1, -2]
dr2 = [1, 1]
dc2 = [-1, 1]

def val(i, j):
    return 0 <= i < 8 and 0 <= j < 8

def getx(in_str):
    return int(in_str[0]) - 1

def gety(in_str):
    return ord(in_str[1]) - ord('a')

c = 1

while True:
    in_str = input().strip()
    if in_str == "0":
        break

    x = getx(in_str)
    y = gety(in_str)

    tab = [[0] * 8 for _ in range(8)]
    for _ in range(8):
        in_str = input().strip()
        tab[getx(in_str)][gety(in_str)] = 1

    ans = 0
    for i in range(8):
        if val(x + dr[i], y + dc[i]):
            flag = 0
            for j in range(2):
                if val(x + dr[i] + dr2[j], y + dc[i] + dc2[j]):
                    if tab[x + dr[i] + dr2[j]][y + dc[i] + dc2[j]]:
                        flag = 1
                        break
            if not flag:
                ans += 1

    print(f"Caso de Teste #{c}: {ans} movimento(s).")
    c += 1

File: AD-HOC/C#/TELA_1/1147.cs
using System;

class Program
{
    static int[] dr = {-1, -2, -2, -1, 1, 2, 2, 1};
    static int[] dc = {-2, -1, 1, 2, 2, 1, -1, -2};
    static int[] dr2 = {1, 1};
    static int[] dc2 = {-1, 1};

    static bool Val(int i, int j)
    {
        return i >= 0 && i < 8 && j >= 0 && j < 8;
    }

    static int GetX(string input)
    {
        return input[0] - '1';
    }

    static int GetY(string input)
    {
        return input[1] - 'a';
    }

    static void Main()
    {
        int[,] tab = new int[8, 8];
        int c = 1;

        while (true)
        {
            string input = Console.ReadLine();
            if (input == "0")
                return;

            int x = GetX(input);
            int y = GetY(input);
            Array.Clear(tab, 0, tab.Length);

            for (int i = 0; i < 8; ++i)
            {
                input = Console.ReadLine();
                tab[GetX(input), GetY(input)] = 1;
            }

            int ans = 0;
            for (int i = 0; i < 8; ++i)
            {
                if (Val(x + dr[i], y + dc[i]))
                {
                    int flag = 0;
                    for (int j = 0; j < 2; ++j)
                    {
                        if (Val(x + dr[i] + dr2[j], y + dc[i] + dc2[j]))
                        {
                            if (tab[x + dr[i] + dr2[j], y + dc[i] + dc2[j]] != 0)
                            {
                                flag = 1;
                                break;
                            }
                        }
                    }
                    if (flag == 0)
                        ++ans;
                }
            }

            Console.WriteLine($"Caso de Teste #{c++}: {ans} movimento(s).");
        }
    }
}